lib/pure/collections/lists.nim

```diff
@@ -752,12 +753,18 @@ proc remove*[T](L: var SinglyLinkedList[T], n: SinglyLinkedNode[T]) =
     var prev = L.head
     while prev.next != n and prev.next != nil:
       prev = prev.next
-    if prev.next != nil:
-      prev.next = n.next
+    if prev.next == nil:
+      return false
+    prev.next = n.next
+  true
 
 proc remove*[T](L: var DoublyLinkedList[T], n: DoublyLinkedNode[T]) =
   ## Removes a node `n` from `L`. Efficiency: O(1).
-  ## When the list is cyclic, the cycle is preserved after removal.
+  ## This function assumes, for the sake of efficiency, that `n` is contained in `L`,
+  ## otherwise the effects are undefined.
+  ## When the list is cyclic, the cycle is preserved after removal,
```

Comment:
can you add a test for that?

Reply:
How would you test undefined effects? :)
If you mean the cycle preservation, I included a link as you recommended...

Reply:
yes i meant cycle preservation